ogre/src/OgreRenderEngine.cc

```diff
@@ -66,6 +70,84 @@ class gz::rendering::OgreRenderEnginePrivate
 using namespace gz;
 using namespace rendering;
 
+// Plugin loading and unloading code adapted from Ogre.
+// Unloading the RenderSystem_GL plugin was found to cause a crash when the
+// rendering thread exits. This only happens on Ubuntu 24.04 when running
+// ogre using system debs, see
+// https://github.com/gazebosim/gz-rendering/issues/1007
+// To workaournd ths issue, we adapth the code for loadng and unloading
+// ogre plugin here. We load the RenderSystem_GL library manually and store it
+// in gz-rendering and on shutdown we stop the plugin and delete the lib without
+// unloading it.
+// \todo(iche033) Find a proper way to unload the library without causing a
+// crash.
+typedef void (*DLL_START_PLUGIN)(void);
```

Comment:
`DLL_START_PLUGIN` has a `const NameValuePairList *` argument in [OgreRoot.cpp:132](https://github.com/OGRECave/ogre-next/blob/v2.3.3/OgreMain/src/OgreRoot.cpp#L132), but is `void` here. Does that matter?

Reply:
Oops thanks!